core/arch/arm32/kernel/tee_ta_manager.c

```diff
@@ -1060,11 +1060,33 @@ static void tee_ta_destroy_context(struct tee_ta_ctx *ctx)
 	free(ctx);
 }
 
+/* check if requester (client ID) matches session initial client */
+static TEE_Result check_client(struct tee_ta_session *s, const TEE_Identity *id)
```

Comment:
It looks like tee_ta_verify_session_pointer() is called in all pathes. I've checked this for "close"